[Bug]: K8s sandbox creation always fails with Jackson >= 2.19 - fabric8 6.13.4 pinned in the BOM is incompatible with the Jackson version the same BOM manages

**Describe the bug**

The dependencies BOM pins `io.fabric8:kubernetes-client` **6.13.4** while the *same* BOM manages Jackson **2.21.1**. fabric8 6.13.x is incompatible with Jackson >= 2.19.0 (fabric8io/kubernetes-client#7036, fixed in fabric8 7.3.0 via fabric8io/kubernetes-client#7038): serializing any object whose `metadata.managedFields[].fieldsV1` is populated crashes with `NullPointerException: keySerializer is null` inside Jackson's `MapSerializer` (via fabric8's `@JsonAnyGetter` handling on `FieldsV1.getAdditionalProperties`).

This makes the Kubernetes sandbox backend unusable out of the box: during sandbox creation, `K8sHelper.resolveSandboxName` watches the `SandboxClaim` CRD. fabric8 6.13.4's `AbstractWatchManager.eventReceived` converts the (untyped) watch event object - a `GenericKubernetesResource` carrying `managedFields` - via `KubernetesSerialization.convertValue(resource, SandboxClaim.class)`, which hits the NPE. The watch is closed with `WatcherException("Unexpected exception processing watch event")`, `future.get()` throws, and sandbox creation fails. `waitForSandboxReady` (`SandboxResource` watch) follows the same pattern.

No user misconfiguration is involved - the broken version combination is what the BOM itself produces. Any application that resolves Jackson to >= 2.19 (which includes every Spring Boot 3.5+/4 app, and AgentScope's own BOM at 2.21.1) is affected.

**To Reproduce**

No cluster needed - the NPE is in serialization, reproducible as a unit test in `agentscope-extensions-sandbox-kubernetes` with the module's own resolved classpath (kubernetes-client 6.13.4 + jackson-databind 2.21.1):

```java
@Test
void convertValueWithManagedFieldsFieldsV1ShouldNotThrow() {
    KubernetesClient client = new KubernetesClientBuilder().build();

    // Mirrors AbstractWatchManager.eventReceived in fabric8 6.13.4:
    // an untyped watch event object (GenericKubernetesResource) is converted
    // to the target custom-resource type.
    GenericKubernetesResource gkr = new GenericKubernetesResource();
    ObjectMeta metadata = new ObjectMeta();
    metadata.setName("sandboxclaim-test");
    ManagedFieldsEntry managedFieldsEntry = new ManagedFieldsEntry();
    FieldsV1 fieldsV1 = new FieldsV1();
    Map<String, Object> additional = new LinkedHashMap<>();
    additional.put("f:metadata", Map.of("f:name", Map.of()));
    fieldsV1.setAdditionalProperties(additional);
    managedFieldsEntry.setFieldsV1(fieldsV1);
    metadata.setManagedFields(List.of(managedFieldsEntry));
    gkr.setMetadata(metadata);

    client.getKubernetesSerialization().convertValue(gkr, SandboxClaim.class); // -> NPE
}
```

Real-world trigger: set `agent.filesystem.mode=SANDBOX` (Kubernetes backend) and run any agent - every sandbox acquisition fails in `resolveSandboxName` (the API server populates `managedFields` by default on Kubernetes >= 1.18, and the claim is always written by at least two managers - the client that creates it and the controller that assigns the sandbox - so the watch event always carries them).

**Expected behavior**

K8s sandbox creation should work with the Jackson version the project itself manages (2.21.1 in the BOM).

**Error messages**

Unit-test reproducer above (verified on current main @ 2.0.3-SNAPSHOT, jackson-databind 2.21.1 + kubernetes-client(-api) 6.13.4):

```
java.lang.IllegalArgumentException: Cannot invoke "JsonSerializer.serialize(Object, JsonGenerator, SerializerProvider)" because "keySerializer" is null (through reference chain: io.fabric8.kubernetes.api.model.GenericKubernetesResource["metadata"]->io.fabric8.kubernetes.api.model.ObjectMeta["managedFields"]->java.util.List[0]->io.fabric8.kubernetes.api.model.ManagedFieldsEntry["fieldsV1"]->io.fabric8.kubernetes.api.model.FieldsV1["getAdditionalProperties"]->java.util.LinkedHashMap["f:metadata"])
Caused by: java.lang.NullPointerException: ... "keySerializer" is null
```

Production stack trace (2.0.1, app on Spring Boot 3.5.14 -> Jackson 2.21.2):

```
SandboxLifecycleMiddleware.acquireForCall
-> SandboxManager.acquire
-> KubernetesSandboxClient.create -> SandboxClient.createSandbox
-> K8sHelper.resolveSandboxName            // future.get() waiting on SandboxClaim watch
-> io.fabric8.kubernetes.client.WatcherException: Unexpected exception processing watch event
-> AbstractWatchManager.eventReceived      // convertValue(GenericKubernetesResource, SandboxClaim.class)
-> KubernetesSerialization.convertValue
-> NPE: keySerializer is null              // FieldsV1["getAdditionalProperties"]
```

**Environment (please complete the following information):**

- AgentScope-Java Version: 2.0.1 (reproduced on current main @ 2.0.3-SNAPSHOT; the same fabric8 6.13.4 + Jackson 2.21.1 combination exists since the 2.0.0 RCs)
- Java Version: 21
- OS: Windows 11 (cluster-side: any Kubernetes >= 1.18, since the API server populates `managedFields` by default)

**Additional context**

- Upstream root cause: fabric8io/kubernetes-client#7036, fixed in fabric8 **7.3.0** (fabric8io/kubernetes-client#7038) - note the fix *requires* Jackson >= 2.19 and is not compatible with older Jackson, so the versions are coupled in both directions: fabric8 <= 6.13 <=> Jackson <= 2.18, fabric8 >= 7.3 <=> Jackson >= 2.19.
- spring-cloud-kubernetes ran into exactly the same wall and resolved it by moving to fabric8 7.3.x: spring-cloud/spring-cloud-kubernetes#1923.
- Suggested fix: bump `fabric8.kubernetes-client.version` in `agentscope-dependencies-bom/pom.xml` from 6.13.4 to >= 7.3.0 (latest is 7.8.0). Verified:
  - the reproducer above passes with fabric8 7.3.0 and 7.8.0 + Jackson 2.21.1;
  - all 39 sources of `agentscope-extensions-sandbox-kubernetes` compile against fabric8 7.8.0 unchanged - no other module uses fabric8, so the blast radius is this one module's version bump plus its test suite.
- Downgrading Jackson to <= 2.18 is not viable long-term: Spring Boot 3.5+/4 (and the ecosystem generally) move Jackson forward, and AgentScope's own BOM already manages 2.21.1.

--- a/agentscope-dependencies-bom/pom.xml
+++ b/agentscope-dependencies-bom/pom.xml
@@ -114,7 +114,7 @@
         <kotlin.version>1.9.24</kotlin.version>
         <kotlin.coroutines.version>1.10.2</kotlin.coroutines.version>
         <jgit.version>7.6.0.202603022253-r</jgit.version>
-        <fabric8.kubernetes-client.version>6.13.4</fabric8.kubernetes-client.version>
+        <fabric8.kubernetes-client.version>7.8.0</fabric8.kubernetes-client.version>
         <protobuf.java.version>4.29.3</protobuf.java.version>
 
         <spotless.version>3.4.0</spotless.version>

--- a/agentscope-extensions/agentscope-extensions-sandbox/agentscope-extensions-sandbox-kubernetes/src/test/java/io/agentscope/extensions/sandbox/kubernetes/KubernetesSerializationCompatTest.java
+++ b/agentscope-extensions/agentscope-extensions-sandbox/agentscope-extensions-sandbox-kubernetes/src/test/java/io/agentscope/extensions/sandbox/kubernetes/KubernetesSerializationCompatTest.java
@@ -1,0 +1,64 @@
+/*
+ * Copyright 2024-2026 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.agentscope.extensions.sandbox.kubernetes;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.agentscope.extensions.sandbox.kubernetes.client.crd.SandboxClaim;
+import io.fabric8.kubernetes.api.model.FieldsV1;
+import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+import io.fabric8.kubernetes.api.model.ManagedFieldsEntry;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Guards against fabric8/Jackson serialization regressions on the CRD watch
+ * path: {@code AbstractWatchManager.eventReceived} converts untyped watch
+ * event objects ({@code GenericKubernetesResource}) to the target custom
+ * resource via {@code KubernetesSerialization.convertValue}. With fabric8
+ * 6.13.x and Jackson >= 2.19 this crashed with a {@code keySerializer is
+ * null} NPE on {@code metadata.managedFields[].fieldsV1} (fabric8 issue
+ * #7036, fixed in fabric8 7.3.0).
+ */
+class KubernetesSerializationCompatTest {
+
+    @Test
+    void convertValueWithManagedFieldsFieldsV1ShouldNotThrow() {
+        try (KubernetesClient client = new KubernetesClientBuilder().build()) {
+            GenericKubernetesResource gkr = new GenericKubernetesResource();
+            ObjectMeta metadata = new ObjectMeta();
+            metadata.setName("sandboxclaim-test");
+
+            ManagedFieldsEntry managedFieldsEntry = new ManagedFieldsEntry();
+            FieldsV1 fieldsV1 = new FieldsV1();
+            Map<String, Object> additional = new LinkedHashMap<>();
+            additional.put("f:metadata", Map.of("f:name", Map.of()));
+            fieldsV1.setAdditionalProperties(additional);
+            managedFieldsEntry.setFieldsV1(fieldsV1);
+            metadata.setManagedFields(List.of(managedFieldsEntry));
+            gkr.setMetadata(metadata);
+
+            SandboxClaim claim =
+                    client.getKubernetesSerialization().convertValue(gkr, SandboxClaim.class);
+            assertEquals("sandboxclaim-test", claim.getMetadata().getName());
+        }
+    }
+}

